stubs/WTForms/wtforms/fields/choices.pyi

```diff
@@ -0,0 +1,78 @@
+from collections.abc import Callable, Iterable, Iterator, Sequence
+from typing import Any
+from typing_extensions import Self, TypeAlias
+
+from wtforms.fields.core import Field, _Filter, _FormT, _Validator, _Widget
+from wtforms.form import BaseForm
+from wtforms.meta import DefaultMeta, _SupportsGettextAndNgettext
+
+# technically this allows a list, but we're more strict for type safety
+_Choice: TypeAlias = tuple[Any, str]
+_GroupedChoices: TypeAlias = dict[str, Iterable[_Choice]]
+_FullChoice: TypeAlias = tuple[Any, str, bool]  # value, label, selected
+_FullGroupedChoices: TypeAlias = tuple[str, Iterable[_FullChoice]]
+
+# this is defined anonymously inside SelectFieldBase, we have to pull it
+# out of there so pytype doesn't crash
+class _Option(Field):
+    checked: bool
+
+class SelectFieldBase(Field):
+    option_widget: _Widget[_Option]
+    def __init__(
+        self,
+        label: str | None = None,
+        validators: tuple[_Validator[_FormT, Self], ...] | list[Any] | None = None,
+        option_widget: _Widget[_Option] | None = None,
+        *,
+        filters: Sequence[_Filter] = (),
+        description: str = "",
+        id: str | None = None,
+        default: object | None = None,
+        widget: _Widget[Self] | None = None,
+        render_kw: dict[str, Any] | None = None,
+        name: str | None = None,
+        _form: BaseForm | None = None,
+        _prefix: str = "",
+        _translations: _SupportsGettextAndNgettext | None = None,
+        _meta: DefaultMeta | None = None,
+    ) -> None: ...
+    def iter_choices(self) -> Iterator[_FullChoice]: ...
+    def has_groups(self) -> bool: ...
+    def iter_groups(self) -> Iterator[_FullGroupedChoices]: ...
+    def __iter__(self) -> Iterator[_Option]: ...
+    _Option: type[_Option]
```

Comment:
Can you use a different name for the class? I'm not sure we can expect it to work reliably when you use the same name here both times.

Reply:
I think it should be fine, since it's the final declaration in the class body. I would like to avoid changing the name, since you actually need to refer to this class sometimes, even though it's undocumented. The only reason it's even declared outside of the class body, is because pytype was crashing if I referred to it by `SelectFieldBase._Option`.

I could maybe try moving the declaration back into the class body, but put it as the very top and see if that solves the crash. The annotations within the class body could also get away with omitting the `SelectFieldBase.` prefix.

Reply:
Looks like moving the declaration to the top does not help pytype out either. Can you think of any other way to preserve the name, that's more robust?

Reply:
I ended up defining a type alias for `type[_Option]` so it can be used for this one declaration, so that should be slightly more robust. Ideally `SelectFieldBase._Option` would be valid as a type, but I'm not sure how to achieve this in a way that will make pytype happy.

Reply:
I came up with a better solution of creating a type alias outside to refer to the class inside, now it doesn't matter if `_Option` refers to the class definition or the type alias inside the class body, the behavior should be the same and pytype is still happy with it too.